### ISSUE TITLE
Error when deleting parent parent directory

### ISSUE THREAD

**Issue by ellishg** [fish-shell/fish-shell#7363]
I found the removing the parent directory is fine, but it seems that if you remove the parent directory of the parent directory of the current directory, I get a very long and annoying error message every time I do a command.

```
$ mkdir -p mydir/myparentdir
$ cd mydir/myparentdir/
$ rm -rf ../../mydir/
pwd: realpath failed:test: Missing argument at index 2

/Users/ellishoag/homebrew/Cellar/fish/3.1.0_1/share/fish/functions/fish_print_hg_root.fish (line 11):
    while test $dir != "/"
          ^
in function 'fish_print_hg_root'
	called on line 1 of file /Users/ellishoag/homebrew/Cellar/fish/3.1.0_1/share/fish/functions/fish_hg_prompt.fish
in command substitution
	called on line 30 of file /Users/ellishoag/homebrew/Cellar/fish/3.1.0_1/share/fish/functions/fish_hg_prompt.fish
in function 'fish_hg_prompt'
	called on line 5 of file /Users/ellishoag/homebrew/Cellar/fish/3.1.0_1/share/fish/functions/fish_vcs_prompt.fish
in function 'fish_vcs_prompt'
	called on line 1 of file /Users/ellishoag/homebrew/Cellar/fish/3.1.0_1/share/fish/functions/fish_prompt.fish
in command substitution
	called on line 28 of file /Users/ellishoag/homebrew/Cellar/fish/3.1.0_1/share/fish/functions/fish_prompt.fish
in function 'fish_prompt'
in command substitution

(Type 'help test' for related documentation)
```

```
$ fish --version
fish, version 3.1.0
$ echo $version
3.1.0
$ uname -a
Darwin ellishoag-mbp 19.6.0 Darwin Kernel Version 19.6.0: Thu Jun 18 20:49:00 PDT 2020; root:xnu-6153.141.1~1/RELEASE_X86_64 x86_64
$ echo $TERM
xterm-256color
```

**Comment by faho** [comment]
Duplicate of #6699

**Comment by faho** [comment]
This was fixed in fish 3.1.1, released in april. Please upgrade.
